lib/fencing/st_actions.c

```diff
@@ -396,6 +396,113 @@ stonith__legacy2status(int rc)
     }
 }
 
+/*!
+ * \internal
+ * \brief Add a fencing result to an XML element as attributes
+ *
+ * \param[in] xml     XML element to add result to
+ * \param[in] result  Fencing result to add (assume success if NULL)
+ */
+void
+stonith__xe_set_result(xmlNode *xml, const pcmk__action_result_t *result)
+{
+    int exit_status = CRM_EX_OK;
+    enum pcmk_exec_status execution_status = PCMK_EXEC_DONE;
+    const char *exit_reason = NULL;
+    const char *action_stdout = NULL;
+    int rc = pcmk_ok;
+
+    CRM_CHECK(xml != NULL, return);
+
+    if (result != NULL) {
+        exit_status = result->exit_status;
```

Comment:
Probably usage later on will reveal why we are defaulting to positive here instead of the rather expected bad and unknown in case of a result == NULL or just not add anything to xml.

Reply:
Yep, that ends up being the common trivial use case -- it avoids the caller having to create a result object